FundTrack.BLL/Concrete/OfferedItemService.cs

```diff
@@ -225,37 +201,40 @@ public OfferedItemViewModel InitializeOfferedItemViewModel(OfferedItem item)
         }
 
         /// <summary>
-        /// Sets specified images of specified offer item
+        ///  Sets specified images of specified offer item. First image is Main
         /// </summary>
-        /// <param name="model">Offer item view model</param>
-        /// <param name="item">Offered item entity</param>
-        /// <returns>List of offered item images</returns>
-        public OfferedItemImageViewModel[] SetNewPictures(OfferedItemViewModel model, OfferedItem item)
+        /// <param name="imagesInBase64code">Base64 code of images</param>
+        /// <param name="offeredItemId">Offered item Id</param>
+        /// <returns>Collection of offered item images</returns>
+        public ICollection<OfferedItemImage> SetNewPictures(string[] imagesInBase64code,int offeredItemId)
         {
-            int i = 0;
-            Dictionary<OfferedItemImage, Task<string>> d = new Dictionary<OfferedItemImage, Task<string>>();
-            foreach (var thing in model.Image)
+            Dictionary<OfferedItemImage, Task<string>> imageTastDictionary = new Dictionary<OfferedItemImage, Task<string>>();
+
+            foreach (var item in imagesInBase64code)
             {
-                var newImage = new OfferedItemImage
+                var newImage = new OfferedItemImage()
                 {
-                    IsMain = thing.IsMain,
-                    OfferedItemId = item.Id
+                    IsMain = false,
+                    OfferedItemId = offeredItemId
                 };
-                var t = _imgService.UploadImage(Convert.FromBase64String(model.Base64Images[i++]));
 
-                d.Add(newImage, t);
+                var t = _imgService.UploadImage(Convert.FromBase64String(item));
+                imageTastDictionary.Add(newImage, t);
             }
-            Task.WhenAll(d.Values);
+            Task.WhenAll(imageTastDictionary.Values);
 
-            foreach (var element in d)
+            foreach (var element in imageTastDictionary)
             {
                 element.Key.ImageUrl = element.Value.Result;
             }
-
-            this._unitOfWork.OfferImagesRepository.CreateMany(d.Keys.ToList());
-            return model.Image;
+            if (imageTastDictionary.Keys.First() != null)
+            {
+                imageTastDictionary.Keys.First().IsMain = true;

```

Comment:
not sure if this is a good place to decide what image is "main". Shouldn't API client (end-user) decide which of the images is the main one? Consider assigniong value of this property on the UI.